lib/_stream_readable.js

```diff
@@ -211,7 +211,7 @@ function howMuchToRead(n, state) {
   if (state.objectMode)
     return n === 0 ? 0 : 1;
 
-  if (n === null || isNaN(n)) {
+  if (n === null || n !== n) {
```

Comment:
@bnoordhuis This does change the functionality from `isNaN(n)` to `Number.isNaN(n)` (don't we all love JS). The only way I can think to make sure it matches is to change the check to:
```js
if (n === null || (+n) != n) {
```